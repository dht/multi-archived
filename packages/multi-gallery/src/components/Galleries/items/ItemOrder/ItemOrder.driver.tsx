import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { ItemOrder, ItemOrderProps } from './ItemOrder';
import { BaseComponentDriver } from 'testing-base';

export class ItemOrderDriver extends BaseComponentDriver {
    private props: Partial<ItemOrderProps> = {};

    constructor() {
        super('ItemOrder');
    }

    when: any = {
        rendered: () => {
            render(<ItemOrder {...(this.props as ItemOrderProps)} />);
            return this;
        },
        clicked: () => {
            fireEvent.click(this.wrapper);
            return this;
        },
        snapshot: () => {
            return this.snapshot(
                <ItemOrder {...(this.props as ItemOrderProps)} />
            );
        },
    };

    given: any = {
        props: (props: Partial<ItemOrderProps>) => {
            this.props = props;
            return this;
        },
    };

    get = {
        containerClassName: () => {
            return this.wrapper.className;
        },
        label: () => {
            return this.wrapper.innerHTML;
        },
    };
}
