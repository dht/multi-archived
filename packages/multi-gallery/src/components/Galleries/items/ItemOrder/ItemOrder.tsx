import React from 'react';
import { Wrapper } from './ItemOrder.style';

export type ItemOrderProps = {};

export function ItemOrder(_props: ItemOrderProps) {
    return (
        <Wrapper className='ItemOrder-wrapper' data-testid='ItemOrder-wrapper'>
            ItemOrder
        </Wrapper>
    );
}

export default ItemOrder;
