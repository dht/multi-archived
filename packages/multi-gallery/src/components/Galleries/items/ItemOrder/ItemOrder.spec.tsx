import { ItemOrderDriver } from './ItemOrder.driver';
import Chance from 'chance';

const chance = new Chance();

describe('ItemOrder', () => {
    let driver: ItemOrderDriver;

    beforeAll(() => {
        driver = new ItemOrderDriver();
    });

    it('should render button', () => {
        const label = chance.word();

        const element = driver.given
            .props({
                title: label,
            })
            .when.rendered();

        const containerClassName = element.get.containerClassName();
        const innerText = element.get.label();

        expect(containerClassName).toContain('ItemOrder-wrapper');
        expect(innerText).toBe(label);
    });

    it('should click button', () => {
        const callback = jest.fn();

        driver.given
            .props({
                onClick: callback,
            })
            .when.rendered()
            .when.clicked();

        expect(callback).toHaveBeenCalledTimes(1);
    });
});

describe('ItemOrder snapshots', () => {
    let driver: ItemOrderDriver;

    beforeAll(() => {
        driver = new ItemOrderDriver();
    });

    it('should match snapshot', () => {
        expect(driver.when.snapshot()).toMatchSnapshot();
    });
});
