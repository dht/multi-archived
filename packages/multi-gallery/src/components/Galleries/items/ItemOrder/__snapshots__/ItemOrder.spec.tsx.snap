// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemOrder snapshots should match snapshot 1`] = `
<div
  className="sc-bczRLJ WtYQR ItemOrder-wrapper"
  data-testid="ItemOrder-wrapper"
>
  ItemOrder
</div>
`;
